### PR TITLE
Fix `pattern AndBool` and friends that did not match the `_andBool_` symbol

### PR DESCRIPTION
**Pull request by geo2a**
Summary of changes:
* correct the pattern synonyms in `Booster.Pattern.Bool` by including the `hook` attribute to the hooked symbols
  - the absence of the attribute in i.e. `AndBool` pattern synonym caused it to *not match* the `_andBool_` symbols from rules's requires clauses;
  - that caused the conjunction splitting functions `splitAndBool` and `splitBoolPredicates` to have no effect, potentially blocking rule applications.
* add unit-tests for these pattern synonyms in `Test.Booster.Pattern.Bool`
  - the test internalize the symbols from surface KORE using the quasi-quoter and test that the internalized symbols are indeed matched by the pattern synonyms.
* change the hook name to be represented as `ByteString` (not `Text`)
  - having both BS and Text causes issues when parsing symbols with both hook and smt-hook attributes using the quasi-quoter
  - that triggered the need to regenerate the responses in `test-foundry-bug-report`

TODO:

- [x] run KEVM performance tests and post results
- [x] run Kontrol performance tests and post results

### CODE REVIEW

**Comment by geo2a** [comment]
**KEVM performance**: no miracle...
| Test                                    | georgy-better-unclear-condition time | master-a956edb time | (georgy-better-unclear-condition/master-a956edb) time
|-----------------------------------------|--------------------------------------|---------------------|-------------------------------------------------------
| mcd/flipper-addu48u48-fail-rough-spec.k | 98.09                                | 111.61              | 0.8788639010841323
| erc20/hkg/approve-spec.k                | 110.39                               | 123.05              | 0.8971149939049168
| erc20/hkg/transferFrom-failure-2-spec.k | 241.72                               | 263.33              | 0.9179356700717731
| mcd/flopper-cage-pass-spec.k            | 178.48                               | 193.87              | 0.9206169082374787
| mcd/vat-mului-pass-spec.k               | 272.78                               | 288.6               | 0.9451836451836451
| mcd/vat-subui-pass-rough-spec.k         | 292.52                               | 307.96              | 0.9498636186517729
| mcd/vat-subui-fail-rough-spec.k         | 148.88                               | 156.27              | 0.9527100531132014
| mcd/cat-file-addr-pass-rough-spec.k     | 190.1                                | 198.48              | 0.9577791213220476
| erc20/hkg/transfer-failure-2-spec.k     | 133.85                               | 139.61              | 0.9587422104433778
| mcd/vat-muluu-pass-spec.k               | 255.08                               | 241.48              | 1.0563193639224782
| benchmarks/storagevar00-spec.k          | 86.91                                | 82.22               | 1.0570420822184383
| mcd/vat-subui-pass-spec.k               | 311.04                               | 290.93              | 1.069123156773107
| mcd/end-subuu-pass-spec.k               | 247.93                               | 230.56              | 1.0753383067314366
| erc20/hkg/balanceOf-spec.k              | 111.47                               | 100.53              | 1.1088232368447228
| TOTAL                                   | 2679.24                              | 2728.5              | 0.9819461242440901

File: library/Booster/Definition/Attributes/Reader.hs [hunk 1]
```diff
@@ -188,7 +193,7 @@ instance HasAttributes ParsedSymbol where
             <*> hasConcreteEvaluators
             <*> pure Nothing
             <*> smt
-            <*> (attributes .:? "hook")
+            <*> hook
```
Comment (jberthold) [inline library/Booster/Definition/Attributes/Reader.hs:196]: ... or maybe just `(encodeUtf8 <$> attributes  .:? "hook")` ? :golfing_man:

File: unit-tests/Test/Booster/Pattern/Bool.hs [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+    testGroup
+        "Pattern synonyms for predicate symbols should match parsed symbol applications"
+        [ testPatternAndBool
+        , testPatternNotBool
+        , testPatternEqualsBool
+        , testPatternEqualsInt
+        , testPatternNEqualsInt
+        , testPatternEqualsK
+        , testPatternNEqualsK
+        , testPatternSetIn
```
Comment (jberthold) [inline unit-tests/Test/Booster/Pattern/Bool.hs:32]: :+1: Very good, covers all hard-wired patterns.
What we could do in addition to this unit test is an integration test which _compiles_ a simple K definition (with `INT` and `BOOL` in scope) that contains specific target terms to match with the hard-wired patterns. That way we would notice changes in the compiler or the `domains.md` prelude.

File: library/Booster/Definition/Attributes/Reader.hs (hunk 1 above)
Comment (geo2a) [reply library/Booster/Definition/Attributes/Reader.hs:196]: It actually needs to be `(fmap encodeUtf8 <$> attributes  .:? "hook")`. I though that was a little too much!

Comment (geo2a) [reply library/Booster/Definition/Attributes/Reader.hs:196]: I've settled on
```
            hook = do
                hooked <- attributes .:? "hook"
                pure (encodeUtf8 <$> hooked)
```

File: unit-tests/Test/Booster/Pattern/Bool.hs (hunk 2 above)
Comment (geo2a) [reply unit-tests/Test/Booster/Pattern/Bool.hs:32]: Good idea, I will do that!

Comment (geo2a) [reply unit-tests/Test/Booster/Pattern/Bool.hs]: I've moved these tests to `test/predicates-integration`.

**Comment by geo2a** [comment]
# Kontrol performance

| Test                                                                   | georgy-better-unclear-condition time | master-0ef6ecd time | (georgy-better-unclear-condition/master-0ef6ecd) time
|------------------------------------------------------------------------|--------------------------------------|---------------------|-------------------------------------------------------
| AssumeTest.testFail_assume_true(uint256,uint256)                       | 68.82                                | 102.85              | 0.6691298006806028
| AssumeTest.test_assume_false(uint256,uint256)                          | 75.75                                | 108.51              | 0.6980923417196572
| src/tests/integration/test_foundry_prove.py::test_foundry_resume_proof | 76.34                                | 108.96              | 0.7006240822320118
| ExpectRevertTest.testFail_expectRevert_multipleReverts()               | 171.95                               | 194.52              | 0.8839707999177462
| EmitContractTest.testExpectEmit()                                      | 131.91                               | 140.62              | 0.9380600199118191
| BMCLoopsTest.test_countdown_concrete()                                 | 24.55                                | 26.16               | 0.9384556574923547
| BytesTypeTest.test_bytes32(bytes32)                                    | 25.22                                | 26.33               | 0.9578427649069503
| src/tests/integration/test_foundry_prove.py::test_foundry_remove_node  | 44.76                                | 43.15               | 1.0373117033603707
| AssertTest.testFail_assert_true()                                      | 39.66                                | 37.87               | 1.0472669659360971
| CounterTest.testSetNumber(uint256)                                     | 247.49                               | 235.03              | 1.0530145087861125
| TOTAL                                                                  | 906.4499999999999                    | 1024.0              | 0.8852050781249999

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -13,7 +13,7 @@ concurrency:
 
 env:
   ghc_version: '9.2.8'
-  hlint_version: '3.4.1'
+  hlint_version: '3.5'
```
Comment (jberthold) [inline .github/workflows/test.yml:16]: o/t: We might want to upgrade to version 3.8?  (and to a newer GHC) at some point...

**Review by jberthold — APPROVED** [review]
LGTM

**Comment by geo2a** [comment]
I've re-run KEVM performance again since both `main` and this branch had changes: 
```
| Test                                     | georgy-better-unclear-condition time | master-0ef6ecd time | (georgy-better-unclear-condition/master-0ef6ecd) time
|------------------------------------------|--------------------------------------|---------------------|-------------------------------------------------------
| erc20/hkg/transferFrom-success-2-spec.k  | 31.67                                | 40.91               | 0.7741383524810561
| benchmarks/storagevar01-spec.k           | 27.12                                | 28.35               | 0.9566137566137566
| erc20/hkg/transfer-failure-2-spec.k      | 27.33                                | 28.4                | 0.9623239436619718
| mcd/dsvalue-read-pass-spec.k             | 72.96                                | 75.78               | 0.962787015043547
| erc20/ds/transferFrom-failure-2-c-spec.k | 28.27                                | 29.34               | 0.9635310156782549
| erc20/hkg/transfer-success-1-spec.k      | 27.38                                | 28.39               | 0.9644240929904896
| erc20/hkg/transfer-failure-1-spec.k      | 26.9                                 | 27.89               | 0.9645034062387952
| benchmarks/requires01-a0gt0-spec.k       | 27.41                                | 28.41               | 0.9648011263639563
| mcd/vat-addui-fail-rough-spec.k          | 103.96                               | 107.69              | 0.9653635435045036
| benchmarks/storagevar03-spec.k           | 27.35                                | 28.32               | 0.9657485875706215
| erc20/hkg/transferFrom-failure-1-spec.k  | 33.22                                | 30.51               | 1.0888233366109472
| bihu/forwardToHotWallet-failure-1-spec.k | 119.25                               | 107.98              | 1.10437117984812
| TOTAL                                    | 552.82                               | 561.97              | 0.9837179920636333
```

**Pull request merged**